ui/src/screens/Profile/components/Edit.tsx

```diff
@@ -0,0 +1,227 @@
+              name="profile_type"
+              defaultChecked={profileType === RECRUITER}
+              onChange={() => {
+                setSelectedProfileType(RECRUITER);
+              }}
+              innerRef={register}
+            >
+              I&apos;m Hiring
+            </Radio>
+          </div>
+          <div className="radio-item">
+            <Radio
+              value={SEEKER}
+              name="profile_type"
+              defaultChecked={profileType === SEEKER}
+              onChange={() => {
+                setSelectedProfileType(SEEKER);
+              }}
+              innerRef={register}
+            >
+              I&apos;m Seeking
+            </Radio>
+          </div>
+        </RadiosHolder>
+      </Fieldset>
+      {selectedProfileType === RECRUITER && (
+        <>
+          <Fieldset>
+            <Label htmlFor="job-title">Job Title</Label>
+            <TextInput
+              id="job-title"
+              name="headline"
+              defaultValue={headline}
+              ref={register}
+              onChange={(e) => setHeadline(e.target.value)}
+            />
+          </Fieldset>
+          <Fieldset>
+            <Label htmlFor="company">Company</Label>
+            <TextInput id="company" name="company" defaultValue={company} ref={register} />
+          </Fieldset>
+          <Fieldset>
+            <Label htmlFor="company-website">Company Website</Label>
+            <TextInput
+              id="company-website"
+              name="website"
+              defaultValue={website}
+              ref={register}
+              onChange={(e) => setWebsite(e.target.value)}
+            />
+            <Hint>
+              Include a company link for potential candiates to learn more about your company
+            </Hint>
+          </Fieldset>
+        </>
+      )}
+      {selectedProfileType === SEEKER && (
+        <>
+          <Fieldset>
+            <Label htmlFor="headline">Headline</Label>
+            <TextInput
+              id="headline"
+              name="headline"
+              defaultValue={headline}
+              ref={register}
+              onChange={(e) => setHeadline(e.target.value)}
+            />
+            <Hint>Give a headline of what you want others to see you as.</Hint>
+          </Fieldset>
+          <Fieldset>
+            <Label htmlFor="seeker-website">Website / Portfolio / Github</Label>
+            <TextInput
+              id="seeker-website"
+              name="website"
+              defaultValue={website}
+              ref={register}
+              onChange={(e) => setWebsite(e.target.value)}
+            />
+            <Hint>Include a link for potential companies to learn more about you.</Hint>
+          </Fieldset>
+        </>
+      )}
+      <Fieldset>
+        <Label htmlFor="email">Email</Label>
+        <TextInput
+          id="email"
+          name="email"
+          defaultValue={email}
+          ref={register({
```

Comment:
Just a suggestion, should we create a Validator class to validate all the things, like Email, Contact No. and put all the logics there?

Reply:
You mean like create a shared object that stores the the validation definitions?

So that we could do something `ref={register(validations.email)}` ?